app/js/browserDetect.js

```diff
@@ -0,0 +1,36 @@
+/* global document, navigator */
+(function () {
+  var supportedBrowserSet = {
+    Chrome: 52,
+    Firefox: 48,
+    Edge: 38.14352,
+    Safari: 10
+  };
+  var userAgent = navigator.userAgent;
+  var temp;
+  var match = userAgent.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [];
+
+  if (/trident/i.test(match[1])) {
+    temp = /\brv[ :]+(\d+)/g.exec(userAgent) || [];
+    return 'IE ' + (temp[1] || '');
+  }
+  if (match[1] === 'Chrome') {
+    temp = userAgent.match(/\b(OPR|Edge)\/(\d+)/);
+    if (temp !== null) return temp.slice(1).join(' ').replace('OPR', 'Opera');
+  }
+  match = match[2] ? [match[1], match[2]] : [navigator.appName, navigator.appVersion, '-?'];
+  if ((temp = userAgent.match(/version\/(\d+)/i)) !== null) {
+    match.splice(1, 1, temp[1]);
+  }
+  if (supportedBrowserSet.hasOwnProperty(match[0]) && supportedBrowserSet[match[0]] <= match[1]) {
```

Comment:
Shouldn't be `supportedBrowserSet[match[0]] > match[1]`?
Also match[1] is String, it might be better to change Number.